### PR TITLE
Add jest with node-fetch example

### PR DESCRIPTION
**Pull request by gribnoysup**
Hi, y'all! 👋

I spent quite some time trying to make Polly properly work with `node-fetch` and `jest` and figured out that this might be helpful to someone else trying to set it up

Not sure if you accept PRs with examples but I figured I'll submit it anyway 😸If you don't feel like this is needed, feel free to close this PR

### CODE REVIEW

File: examples/jest-node-fetch/__tests__/index.test.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      }
+    }
+  });
+
+  afterAll(async () => {
+    await polly.stop();
+  });
+
+  describe('posts', () => {
+    it('should return post', async () => {
```
[the remark this answers is not in the document]
Comment (gribnoysup) [reply examples/jest-node-fetch/__tests__/index.test.js]: @offirgolan thanks for the feedback! I will totally update that, but I have a few questions first:

1. Would it make sense then to separate tests completely and re-init Polly before each test or this is not considered a common/good approach with other test runners?
2. Is it considered the bad practice to have Polly records under the same name for multiple tests? That one is more like a general question about Polly, I guess 😅

**Comment by gribnoysup** [comment]
Went ahead and refactored the example to re-init Polly before each test run. Now every test recording has its own name 👍Hope this approach is in line with Polly's idea of how it can be done 😸

File: examples/jest-node-fetch/__tests__/index.test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (offirgolan) [reply examples/jest-node-fetch/__tests__/index.test.js]: > Would it make sense then to re-init Polly before each test or this is not considered a common/good approach with other test runners?

I recommend creating a new polly instance for each test, that way each test is completely isolated from the next. An alternative to what you have is using `beforeEach` and the `afterEach` hooks to create a new instance and each test can change the recordingName before doing anything else. 

> Is it considered the bad practice to have Polly records under the same name for multiple tests?

Yeah, I would say so. Creating a recording for each test will make sure that each test is completely isolated from the next. If there are specific recorded requests you wish to share between tests/recordings, you can use the server's [recordingName()] (https://netflix.github.io/pollyjs/#/server/route-handler?id=recordingname) api.

**Comment by offirgolan** [comment]
Thank you for making the changes! What you have here looks great and I think it will be super useful to others new to the library.

File: examples/jest-node-fetch/__tests__/index.test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (gribnoysup) [reply examples/jest-node-fetch/__tests__/index.test.js]: Thanks a lot for answering and explaining the best approach!

**Pull request merged**